### PR TITLE
Add new values for direction prop

### PR DESCRIPTION
**Pull request by riourbana**
I needed a right Sidebar, so I thought it would be a good idea to change the logic of the _direction_ prop in order to support top, bottom, left and right positioning.

Now _direction_ accepted values are _top_, _bottom_, _left_ or _right_  (defaults to _bottom_) instead of _x_/_y_.

### CODE REVIEW

**Comment by hanford** [comment]
Wow thanks for the PR @riourbana ! I'll take a look either today or this weekend

File: src/index.js [hunk 1]
```diff
@@ -298,7 +371,7 @@ export default class Drawer extends Component {
           return (
             <div
               id={id}
-              style={{backgroundColor: `rgba(55, 56, 56, ${open ? 0.6 : 0})`}}
+              style={{ backgroundColor: `rgba(55, 56, 56, ${open ? 0.6 : 0})`, overflowX: 'hidden' }}
```
Comment (hanford) [inline src/index.js]: is this needed? if it is needed, is it only needed when `isDirectionLeft` or `isDirectionRight` .. if that is the case, maybe we should only apply it when one of those is truthy

File: src/index.js [hunk 2]
```diff
@@ -164,7 +176,13 @@ export default class Drawer extends Component {
       this.MOVING_POSITION = movingPosition
       this.NEW_POSITION = newPosition
 
-      if (newPosition >= 0 && this.shouldWeCloseDrawer()) {
+      var positionThreshold = 0;
```
Comment (hanford) [inline src/index.js]: use `let` instead of `var`

File: src/index.js [hunk 3]
```diff
@@ -194,31 +212,45 @@ export default class Drawer extends Component {
     if (this.shouldWeCloseDrawer()) {
       this.hideDrawer()
     } else {
+      var newPosition = 0;
```
Comment (hanford) [inline src/index.js]: use let instead of var

File: src/index.js [hunk 4]
```diff
@@ -194,31 +212,45 @@ export default class Drawer extends Component {
...
     } else {
       this.NEGATIVE_SCROLL = size - element.scrollWidth - this.MAX_NEGATIVE_SCROLL
     }
   }
 
   hideDrawer = () => {
-    const { allowClose, onRequestClose } = this.props
+    const { allowClose, onRequestClose, direction } = this.props
+
+    var defaultPosition = 0
```
Comment (hanford) [inline src/index.js]: use let instead of var

File: src/index.js [hunk 5]
```diff
@@ -238,24 +270,62 @@ export default class Drawer extends Component {
   }
 
   shouldWeCloseDrawer = () => {
-    const { start } = this.state
+    const { start: touchStart } = this.state
+    const { direction } = this.props;
 
-    if (this.MOVING_POSITION === 0) return false
+    var initialPosition = 0
```
Comment (hanford) [inline src/index.js]: use let instead of var

File: src/index.js (hunk 1 above)
Comment (riourbana) [reply src/index.js]: Yes, it is needed only when `isDirectionRight` is true to hide the x-scrollbar during the sliding animation. We can set it to the default value for all the other directions.

File: src/index.js [hunk 6]
```diff
@@ -371,7 +371,7 @@ export default class Drawer extends Component {
           return (
             <div
               id={id}
-              style={{ backgroundColor: `rgba(55, 56, 56, ${open ? 0.6 : 0})`, overflowX: 'hidden' }}
+              style={{ backgroundColor: `rgba(55, 56, 56, ${open ? 0.6 : 0})`, overflowX: isDirectionRight(direction) ? 'hidden' : 'visible'}}
```
Comment (hanford) [inline src/index.js]: might be easier to just construct that style object in the render function like this

```
let styles = {
	backgroundColor: `rgba(55, 56, 56, ${open ? 0.6 : 0})`,
}

if (isDirectionRight(direction)) {
styles = {
 ...styles,
`overflowX: 'hidden'
}

```

Comment (hanford) [reply src/index.js]: that way there is no fingerprint on the style object for people who aren't using this option.

does that make sense?

**Comment by hanford** [comment]
This is super close to getting merged. Thanks again! Can't wait 😬

File: src/index.js (hunk 6 above)
Comment (riourbana) [reply src/index.js]: You're right, maybe it's prettier :) it should be ok now. Thank you @hanford

**Comment by hanford** [comment]
I'm going to go ahead and merge this. Going to fix up some of the styles that got a little out of whack in your more recent commit, but nbd I can fix it.

**Comment by hanford** [comment]
These changes are live in version `3.3.0`!

**Pull request merged**